### ISSUE TITLE
Initial account registration fails.

### ISSUE THREAD

**Issue by dalso0418** [kouprlabs/voltaserve#115]
As in the example, when you set up the .env and docker-compose.yml files to create a container
Web access is possible, but the following error occurs when creating an account.

![image](https://github.com/kouprlabs/voltaserve/assets/107911055/d05fe22e-0843-4d92-a266-c1526f7fefb1)

voltaserve-meilisearch-1  | 2024-07-05T09:15:49.463378Z  INFO HTTP request{method=GET host="127.0.0.1:7700" route=/health query_parameters= user_agent=curl/8.5.0 status_code=200}: meilisearch: close time.busy=543µs time.idle=19.1µs
voltaserve-idp-1          | 521 |         result = new this._Promise((resolve, reject) => {
voltaserve-idp-1          | 522 |           query.callback = (err, res) => (err ? reject(err) : resolve(res))
voltaserve-idp-1          | 523 |         }).catch((err) => {
voltaserve-idp-1          | 524 |           // replace the stack trace that leads to `TCP.onStreamRead` with one that leads back to the
voltaserve-idp-1          | 525 |           // application that created the query
voltaserve-idp-1          | 526 |           Error.captureStackTrace(err)
voltaserve-idp-1          |                       ^
voltaserve-idp-1          | error: relation "user" does not exist
voltaserve-idp-1          |  code: "42P01"
voltaserve-idp-1          | 
voltaserve-idp-1          |       at /app/node_modules/pg/lib/client.js:526:17
voltaserve-idp-1          |       at processTicksAndRejections (native:1:1)
voltaserve-idp-1          | 
voltaserve-idp-1          | POST /v2/accounts - 30.180 ms - -
voltaserve-idp-1          | Error: relation "user" does not exist
voltaserve-idp-1          |     at <anonymous> (/app/node_modules/pg/lib/client.js:526:17)
voltaserve-idp-1          |     at processTicksAndRejections (native)
voltaserve-meilisearch-1  | 2024-07-05T09:16:19.518755Z  INFO HTTP request{method=GET host="127.0.0.1:7700" route=/health query_parameters= user_agent=curl/8

**Comment by bouassaba** [comment]
@dalso0418 I have a question regarding the steps you followed, does it look like this?
1. Clone the Git repository: `git clone https://github.com/kouprlabs/voltaserve.git`
2. While inside the repository folder, download and start the containers: `docker-compose up -d`

**Comment by loboda4450** [comment]
This issue is related to wrongly created tables in database. Try adding this queries [here](https://github.com/kouprlabs/voltaserve/blob/main/postgres/schema.sql#L10)

```sql
CREATE USER voltaserve CREATEDB NOCREATEROLE;
SET ROLE voltaserve;
CREATE DATABASE voltaserve;
SET ROLE root;
ALTER USER voltaserve NOCREATEDB;
GRANT ALL ON DATABASE voltaserve TO voltaserve;
SET ROLE voltaserve;
```

**Comment by bouassaba** [comment]
@dalso0418 could you please confirm if the issue has been solved on your side? Thanks.

**Comment by stancubed** [comment]
I'm also having this issue and getting `relation "user" does not exist`. I tried connecting directly to postgres and running the above queries but it hasn't resolved the issue.

```sql
error: relation "user" does not exist
    at /app/node_modules/pg/lib/client.js:526:17
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async UserRepoImpl.isUsernameAvailable (/app/src/user/repo.ts:131:26)
    at async createUser (/app/src/account/service.ts:53:9)
    at async /app/src/account/router.ts:48:16 {
  length: 103,
  severity: 'ERROR',
  code: '42P01',
  detail: undefined,
  hint: undefined,
  position: '15',
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'parse_relation.c',
  line: '1449',
  routine: 'parserOpenTable'
}
POST /v2/accounts 500 13.324 ms - 201
error: relation "user" does not exist
    at /app/node_modules/pg/lib/client.js:526:17
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async UserRepoImpl.isUsernameAvailable (/app/src/user/repo.ts:131:26)
    at async createUser (/app/src/account/service.ts:53:9)
    at async /app/src/account/router.ts:48:16
```

I ran the following command to list all tables since I didn't see them in postgres and it seems like the tables weren't created?
```sql
SELECT table_schema,table_name
FROM information_schema.tables
ORDER BY table_schema,table_name;
```
Results:
```
information_schema	_pg_foreign_data_wrappers
information_schema	_pg_foreign_servers
information_schema	_pg_foreign_table_columns
information_schema	_pg_foreign_tables
information_schema	_pg_user_mappings
information_schema	administrable_role_authorizations
information_schema	applicable_roles
information_schema	attributes
information_schema	character_sets
information_schema	check_constraint_routine_usage
information_schema	check_constraints
information_schema	collation_character_set_applicability
information_schema	collations
information_schema	column_column_usage
information_schema	column_domain_usage
information_schema	column_options
information_schema	column_privileges
information_schema	column_udt_usage
information_schema	columns
information_schema	constraint_column_usage
information_schema	constraint_table_usage
information_schema	data_type_privileges
information_schema	domain_constraints
information_schema	domain_udt_usage
information_schema	domains
information_schema	element_types
information_schema	enabled_roles
information_schema	foreign_data_wrapper_options
information_schema	foreign_data_wrappers
information_schema	foreign_server_options
information_schema	foreign_servers
information_schema	foreign_table_options
information_schema	foreign_tables
information_schema	information_schema_catalog_name
information_schema	key_column_usage
information_schema	parameters
information_schema	referential_constraints
information_schema	role_column_grants
information_schema	role_routine_grants
information_schema	role_table_grants
information_schema	role_udt_grants
information_schema	role_usage_grants
information_schema	routine_column_usage
information_schema	routine_privileges
information_schema	routine_routine_usage
information_schema	routine_sequence_usage
information_schema	routine_table_usage
information_schema	routines
information_schema	schemata
information_schema	sequences
information_schema	sql_features
information_schema	sql_implementation_info
information_schema	sql_parts
information_schema	sql_sizing
information_schema	table_constraints
information_schema	table_privileges
information_schema	tables
information_schema	transforms
information_schema	triggered_update_columns
information_schema	triggers
information_schema	udt_privileges
information_schema	usage_privileges
information_schema	user_defined_types
information_schema	user_mapping_options
information_schema	user_mappings
information_schema	view_column_usage
information_schema	view_routine_usage
information_schema	view_table_usage
information_schema	views
pg_catalog	pg_aggregate
pg_catalog	pg_am
pg_catalog	pg_amop
pg_catalog	pg_amproc
pg_catalog	pg_attrdef
pg_catalog	pg_attribute
pg_catalog	pg_auth_members
pg_catalog	pg_authid
pg_catalog	pg_available_extension_versions
pg_catalog	pg_available_extensions
pg_catalog	pg_backend_memory_contexts
pg_catalog	pg_cast
pg_catalog	pg_class
pg_catalog	pg_collation
pg_catalog	pg_config
pg_catalog	pg_constraint
pg_catalog	pg_conversion
pg_catalog	pg_cursors
pg_catalog	pg_database
pg_catalog	pg_db_role_setting
pg_catalog	pg_default_acl
pg_catalog	pg_depend
pg_catalog	pg_description
pg_catalog	pg_enum
pg_catalog	pg_event_trigger
pg_catalog	pg_extension
pg_catalog	pg_file_settings
pg_catalog	pg_foreign_data_wrapper
pg_catalog	pg_foreign_server
pg_catalog	pg_foreign_table
pg_catalog	pg_group
pg_catalog	pg_hba_file_rules
pg_catalog	pg_ident_file_mappings
pg_catalog	pg_index
pg_catalog	pg_indexes
pg_catalog	pg_inherits
pg_catalog	pg_init_privs
pg_catalog	pg_language
pg_catalog	pg_largeobject
pg_catalog	pg_largeobject_metadata
pg_catalog	pg_locks
pg_catalog	pg_matviews
pg_catalog	pg_namespace
pg_catalog	pg_opclass
pg_catalog	pg_operator
pg_catalog	pg_opfamily
pg_catalog	pg_parameter_acl
pg_catalog	pg_partitioned_table
pg_catalog	pg_policies
pg_catalog	pg_policy
pg_catalog	pg_prepared_statements
pg_catalog	pg_prepared_xacts
pg_catalog	pg_proc
pg_catalog	pg_publication
pg_catalog	pg_publication_namespace
pg_catalog	pg_publication_rel
pg_catalog	pg_publication_tables
pg_catalog	pg_range
pg_catalog	pg_replication_origin
pg_catalog	pg_replication_origin_status
pg_catalog	pg_replication_slots
pg_catalog	pg_rewrite
pg_catalog	pg_roles
pg_catalog	pg_rules
pg_catalog	pg_seclabel
pg_catalog	pg_seclabels
pg_catalog	pg_sequence
pg_catalog	pg_sequences
pg_catalog	pg_settings
pg_catalog	pg_shadow
pg_catalog	pg_shdepend
pg_catalog	pg_shdescription
pg_catalog	pg_shmem_allocations
pg_catalog	pg_shseclabel
pg_catalog	pg_stat_activity
pg_catalog	pg_stat_all_indexes
pg_catalog	pg_stat_all_tables
pg_catalog	pg_stat_archiver
pg_catalog	pg_stat_bgwriter
pg_catalog	pg_stat_database
pg_catalog	pg_stat_database_conflicts
pg_catalog	pg_stat_gssapi
pg_catalog	pg_stat_io
pg_catalog	pg_stat_progress_analyze
pg_catalog	pg_stat_progress_basebackup
pg_catalog	pg_stat_progress_cluster
pg_catalog	pg_stat_progress_copy
pg_catalog	pg_stat_progress_create_index
pg_catalog	pg_stat_progress_vacuum
pg_catalog	pg_stat_recovery_prefetch
pg_catalog	pg_stat_replication
pg_catalog	pg_stat_replication_slots
pg_catalog	pg_stat_slru
pg_catalog	pg_stat_ssl
pg_catalog	pg_stat_subscription
pg_catalog	pg_stat_subscription_stats
pg_catalog	pg_stat_sys_indexes
pg_catalog	pg_stat_sys_tables
pg_catalog	pg_stat_user_functions
pg_catalog	pg_stat_user_indexes
pg_catalog	pg_stat_user_tables
pg_catalog	pg_stat_wal
pg_catalog	pg_stat_wal_receiver
pg_catalog	pg_stat_xact_all_tables
pg_catalog	pg_stat_xact_sys_tables
pg_catalog	pg_stat_xact_user_functions
pg_catalog	pg_stat_xact_user_tables
pg_catalog	pg_statio_all_indexes
pg_catalog	pg_statio_all_sequences
pg_catalog	pg_statio_all_tables
pg_catalog	pg_statio_sys_indexes
pg_catalog	pg_statio_sys_sequences
pg_catalog	pg_statio_sys_tables
pg_catalog	pg_statio_user_indexes
pg_catalog	pg_statio_user_sequences
pg_catalog	pg_statio_user_tables
pg_catalog	pg_statistic
pg_catalog	pg_statistic_ext
pg_catalog	pg_statistic_ext_data
pg_catalog	pg_stats
pg_catalog	pg_stats_ext
pg_catalog	pg_stats_ext_exprs
pg_catalog	pg_subscription
pg_catalog	pg_subscription_rel
pg_catalog	pg_tables
pg_catalog	pg_tablespace
pg_catalog	pg_timezone_abbrevs
pg_catalog	pg_timezone_names
pg_catalog	pg_transform
pg_catalog	pg_trigger
pg_catalog	pg_ts_config
pg_catalog	pg_ts_config_map
pg_catalog	pg_ts_dict
pg_catalog	pg_ts_parser
pg_catalog	pg_ts_template
pg_catalog	pg_type
pg_catalog	pg_user
pg_catalog	pg_user_mapping
pg_catalog	pg_user_mappings
pg_catalog	pg_views
```

**Comment by stancubed** [comment]
Ah, you know as soon as I posted this, I figured it out. To put this behind my reverse proxy, I put all of the containers on a network and only add the containers that need to be reached from the outside to the shared bridge network with my reverse proxy. I hadn't added a network to the `migrations` container because it didn't have any exposed ports. So it couldn't see `postgres` and was panicking due to connection errors.

So @dalso0418 maybe check the logs from `migrations` and see if it wasn't able to connect and apply the SQL migrations?
